src/multibody/liegroup/liegroup-variant-visitors.hxx

```diff
@@ -39,6 +39,24 @@ namespace pinocchio
         return boost::apply_visitor(Visitor(args),lg);
       }
     };
+
+    struct LieGroupEqual : public boost::static_visitor<bool>
+    {
+      template <typename T, typename U>
+      bool operator()( const T &, const U & ) const
+      {
+        // Should we handle
+        // - dynamic size vector space versus static size vector space
+        // - dynamic versus static cartesian product
```

Comment:
Please, pay attention to this comment.

Reply:
I think we should handle the first. For the second, I don't think it is useful.

Reply:
I think everything should be dynamic, as a Model is dynamic by construction.

Reply:
I think it is relevant to have R1, R3, R3xSO3 and R2xSO2 static.

Reply:
Yes, of course.

Reply:
So both questions make sense:
- dynamic(R^n, 3) == R^3 ?
- CartesianProductVariant(R^3, SO3) == R3xSO3 ?

Reply:
I would say yes.

Reply:
But, for instance, for JointModelVariants, we are doing this, but we should while it is a bit more complex.

Reply:
I don't know for JointModelVariant but here, I don't think it is very complex.

Reply:
Yes, I do agree.